mathoxide-lib/src/array.rs

```diff
@@ -1,15 +1,110 @@
 use crate::storage::Storage;
+use crate::views;
+use crate::views::ArrayView;
 
-pub struct Array<StorageType> {
+use num_traits::Num;
+
+use std::fmt;
+
+pub struct Array<StorageType, ViewType> {
     storage: StorageType,
+    view: ViewType,
+}
+
+impl<T, StorageType, ViewType> Array<StorageType, ViewType>
+where
+    T: Num,
+    StorageType: Storage<Stored = T>,
+    ViewType: ArrayView,
+{
+    fn offset(&self) -> usize {
+        self.view.offset()
+    }
+    // fn stride(&self) -> &[usize];
+    fn ndims(&self) -> usize {
+        self.view.ndims()
+    }
+
+    fn size(&self) -> usize {
+        self.view.size()
+    }
+
+    fn shape(&self) -> &[usize] {
+        self.view.shape()
+    }
+}
+
+impl<T, StorageType, ViewType> fmt::Display for Array<StorageType, ViewType>
+where
+    T: Num + fmt::Display,
+    StorageType: Storage<Stored = T>,
+    ViewType: ArrayView,
+{
+    // TODO: Rewrite this when more helper methods are available
+    // Try to match numpy's behavior for ndarray formatting
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = String::new();
+        match self.storage.storage_get() {
+            Ok(arr) => match self.ndims() {
+                2 => {
```

Comment:
The arms of these `match` statements get a bit long. Perhaps in a next PR a `StorageFormatter` struct could be defined that holds a reference to a slice, and a reference to a view, and internally handles the formatting. So then this `Display` implementation could be written something like:

```rust
fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
    match self.storage.storage_get() {
        Ok(arr) => StorageFormatter::new(arr, self.view).format(),
        Err(err) => write!(f, "error while formatting array: {}", err),
    }
}
```
And the actuall implementation will be moved in the new struct. This might come handy when we actually write a complete implementation of array formatting.